parsers/mcmaster-parser-2.js

```diff
@@ -19,6 +19,50 @@ function convertTo24Hour(time) {
 }
 
 /**
+ * Takes any date and turns it into the correct format, so javascript gets the right date.
+ * MM/DD/YYYY is the format the javascript object reads or YYYY/MM/DD
+ * @param {String} startDate
+ * @param {String} endDate
+ * @return {String Array} [startDate, endDate]  with format MM/DD/YYYY or YYYY/MM/DD
+ */
+function fixDateFormat(sem_start, sem_end) {
+    if (sem_start.indexOf('-') > -1) {
+        return [sem_start,sem_end]
+    }
+    //assume format is either YYYY/MM/DD TO START OR MM/DD/YYYY
+    startDateSplit = sem_start.split("/");
+    endDateSplit = sem_end.split("/");
+
+    startMonth = parseInt(startDateSplit[0]);
+    startDay = parseInt(startDateSplit[1]);
+    startYear = parseInt(startDateSplit[2]);
+
+    endMonth = parseInt(endDateSplit[0]);
+    endDay = parseInt(endDateSplit[1]);
+    endYear = parseInt(endDateSplit[2]);
+
+    // It is YYYY/MM/DD, because the 0th index is a year or
+    //If the difference is 3 months (Fall/Winter), 1 month (Spring/Summer) or -5 months (multi-semester(Sept-Apr))
+    // and the semester starts in [1,5,7,9], Check is month is Jan,May,July or Sept
+    if(startMonth.toString().length === 4 ||
+        ((endMonth - startMonth === 3 || endMonth - startMonth === 1 || endMonth - startMonth === -5) && [1,5,7,9].indexOf(startMonth) > -1)){
+        return [sem_start, sem_end]
+    } else if (startMonth > 12 || endMonth > 12 || (startYear === endYear && startMonth > endMonth)) {
+        // Date is DD/MM/YYYY because month can't be > 12, so switch the month and day around
+        return [[startDay, startMonth, startYear].join("/"),[endDay, endMonth, endYear].join("/")];
+
+        //If the difference is 3 months (Fall/Winter), 1 month (Spring/Summer) or -5 months (multi-semester(Sept-Apr)),
+        // in this case the day is in the month slot and the semester starts in [1,5,7,9], Check is month is Jan,May,July or Sept
+    } else if ((endDay - startDay === 3 || endDay - startDay === 1 || endDay - startDay === -5) && [1,5,7,9].indexOf(startDay) > -1) {
+        // DD/MM/YYYY is the format found because the DD index [1] slot is a Month not a day, so switch the month and day around
+        return [[startDay, startMonth, startYear].join("/"),[endDay, endMonth, endYear].join("/")]
+    }
+
+    //Default
```

Comment:
remove this comment and the line above it, we already know it's the default because there's if statements above it that return other things